joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: &loop_rate 100

whole_body_controller: &whole_body_controller
  type: effort_controllers/JointTrajectoryController  
  joints:
    - torso_lift_joint
    - r_upper_arm_roll_joint
    - r_shoulder_pan_joint
    - r_shoulder_lift_joint
    - r_forearm_roll_joint
    - r_elbow_flex_joint
    - r_wrist_flex_joint
    - r_wrist_roll_joint
    - l_upper_arm_roll_joint
    - l_shoulder_pan_joint
    - l_shoulder_lift_joint
    - l_forearm_roll_joint
    - l_elbow_flex_joint
    - l_wrist_flex_joint
    - l_wrist_roll_joint
    - head_pan_joint
    - head_tilt_joint
  gains:
    torso_lift_joint: {p: 2000, i: 100, d: 50, i_clamp: 10}
    r_upper_arm_roll_joint: {p: 2000, i: 100, d: 50, i_clamp: 10}
    r_shoulder_pan_joint: {p: 2000, i: 100, d: 50, i_clamp: 10}
    r_shoulder_lift_joint: {p: 2000, i: 100, d: 50, i_clamp: 10}
    r_forearm_roll_joint: {p: 2000, i: 100, d: 50, i_clamp: 10}
    r_elbow_flex_joint: {p: 2000, i: 100, d: 50, i_clamp: 10}
    r_wrist_flex_joint: {p: 2000, i: 100, d: 50, i_clamp: 10}
    r_wrist_roll_joint: {p: 2000, i: 100, d: 50, i_clamp: 10}
    l_upper_arm_roll_joint: {p: 2000, i: 100, d: 50, i_clamp: 10}
    l_shoulder_pan_joint: {p: 2000, i: 100, d: 50, i_clamp: 10}
    l_shoulder_lift_joint: {p: 2000, i: 100, d: 50, i_clamp: 10}
    l_forearm_roll_joint: {p: 2000, i: 100, d: 50, i_clamp: 10}
    l_elbow_flex_joint: {p: 2000, i: 100, d: 50, i_clamp: 10}
    l_wrist_flex_joint: {p: 2000, i: 100, d: 50, i_clamp: 10}
    l_wrist_roll_joint: {p: 2000, i: 100, d: 50, i_clamp: 10}
    head_pan_joint: {p: 2000, i: 100, d: 50, i_clamp: 10}
    head_tilt_joint: {p: 2000, i: 100, d: 50, i_clamp: 10}
  constraints:
    stopped_velocity_tolerance: 0

l_gripper_l_finger_controller:
  type: effort_controllers/JointTrajectoryController
  joints: 
    - l_gripper_l_finger_joint
  gains: 
    l_gripper_l_finger_joint: { p: 2000, i: 100, d: 50 }

r_gripper_l_finger_controller:
  type: effort_controllers/JointTrajectoryController
  joints: 
    - r_gripper_l_finger_joint
  gains: 
    r_gripper_l_finger_joint: { p: 2000, i: 100, d: 50 }